src/main/java/org/motechproject/mots/management/LocationImporter.java

```diff
@@ -0,0 +1,206 @@
+package org.motechproject.mots.management;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import org.apache.poi.xssf.usermodel.XSSFCell;
+import org.apache.poi.xssf.usermodel.XSSFRow;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.motechproject.mots.domain.Chiefdom;
+import org.motechproject.mots.domain.Community;
+import org.motechproject.mots.domain.District;
+import org.motechproject.mots.domain.Facility;
+import org.motechproject.mots.service.LocationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+@Component
+public class LocationImporter implements ApplicationRunner {
+
+  private LocationService locationService;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(LocationImporter.class);
+
+  @Value("${mots.loadLocations}")
+  private boolean loadLocations;
+
+  private static final String DISTRICT_HEADER = "District";
+  private static final String CHIEFDOM_HEADER = "Chiefdom";
+  private static final String FACILITY_HEADER = "Health Facility";
+  private static final String COMMUNITY_HEADER = "Community";
+
+
+  private List<District> districtList;
+  private List<Chiefdom> chiefdomList;
+  private List<Facility> facilityList;
+  private List<Community> communityList;
+
+  /**
+   * Initializes locationService and locations lists.
+   * @param locationService service to autowire.
+   */
+  @Autowired
+  public LocationImporter(LocationService locationService) {
+    this.locationService = locationService;
+    if (loadLocations) {
```

Comment:
move this to the run method, the loadLocations field is set after this autowire is executed (at least on my machine)

Reply:
hm works differently on mine, I'll move it since there is no difference